src/parse.rs

```diff
@@ -163,33 +171,55 @@ struct CompilerContext<'pt> {
 impl<'pt> CompilerContext<'pt> {
     fn new(grm: &'pt YaccGrammar, input: &'pt str) -> CompilerContext<'pt> {
         CompilerContext {
-            classes: HashMap::new(),
             symbols: HashMap::new(),
+            bytecode: Vec::new(),
+            labels: HashMap::new(),
             grm:     grm,
             input:   input,
             cur_cls: "global".to_string(),
             cur_fn:  "global".to_string(),
         }
     }
 
+    // The bytecode length of the current class.
+    // Useful for building jump labels
+    fn get_bytecode_length(&self) -> usize {
+        self.bytecode.len()
+    }
+
+    // Used when building up conditional branches and loops, where the pos. to
+    // jump to is not known until all the relevant code is generated.
+    fn patch(&mut self, pos: usize) {
+        let patch_value = self.bytecode.len();
+        let ref mut jump_instr = self.bytecode[pos];
+        match *jump_instr {
+            Instr::JUMP_IF_TRUE(ref mut _i) => { *_i = patch_value; },
```

Comment:
You probably don't want the curlies around the RHS expression.

Reply:
fixed